commands/clans.js

```diff
@@ -11,18 +12,18 @@ module.exports.execute = async (client, message) => {
 	await Promise.all(clans.map(async clan => {
 	//await clans.reduce(async (memo, clan) => { // ensure they return in the order of clan-data
 		//await memo;
-		api = new Habitica({
+		let api = new Habitica({
 			id: clan.id,
 			apiToken: clan.apiToken
 		});
-		clanInfo = await api.get(`/groups/party`).then(res => {return res.data;})
+		let clanInfo = await api.get(`/groups/party`).then(res => {return res.data;})
 			.catch(err => {console.log(`There has been a problem in fetching clan ${clan.fullName}: ${err}`);});
-		// TODO: read in leader's discord id using database, if not exist then use habitica userName
-		clanLeader = clanInfo.leader.profile.name;//id;
-		clanLeaderDescriptor = '0000';
-		memberCount = clanInfo.memberCount;
-		response += `🔸 **${clanInfo.name}**: ${clan.description} by **${clanLeader}**#${clanLeaderDescriptor}\n`
-		response += `        Current members: ${memberCount}/30\n`
+		// find the leader's discord id using database, if not exist then use habitica userName
+		let clanLeader = await habHelper.findDiscordUser(client, clanInfo.leader.id);// TODO: deal with the case that there are two user having same habitica ID
```

Comment:
I'm not sure there's a case where someone can have the same ID, unless it's tracking by name rather than username or token.

Reply:
Re the data file: commands/clan-data-auth-sample is an **example** of the file needed to initialize the db, the actual file that is required is commands/clan-data-auth (I have sent you the file via dm). The clan-data file is still needed for looking up clan names as array is not an allowed datatype for sqlite.

Also I found that the code was a bit out of date since I changed the data files: the next commit is the updated version.

Re the case of having same id: this is referring to the case that someone set their habitica id as someone else's (as habitica uuid is public). Authentication with token could solve this problem but that might make people less willing to link their habitica id.